### PR TITLE
Add JsonPatch.EnumerateArray API with propagator append merge

### PR DESCRIPTION
**Pull request by m-nash**
## Summary
Adds `JsonPatch.EnumerateArray(ReadOnlySpan<byte> jsonPath)` - a new API that iterates over JSON array elements at a given path, yielding each element as `ReadOnlyMemory<byte>`.

### Usage
```csharp
JsonPatch jp = GetFromSomewhere();
foreach (ReadOnlyMemory<byte> item in jp.EnumerateArray("$.foo.array"u8))
{
    // process each array element
}
```

### New files
- **JsonPatch.Iterate.cs** - EnumerateArray, ResolveArray, ApplySubPathMutations, ArrayEnumerator
- **AvailabilitySetDataV2.cs/.Serialization.cs** - V2 test model with enhanced PropagateGet
- **JsonPatchIterateTests.cs** - 61 tests across 7 categories

### Modified files
- **JsonPatch.Implementation.cs** - Merge ArrayItemAppend with propagator results in TryGetEncodedValueInternal
- **TestClientModelReaderWriterContext.cs** - Register V2 builder

## Test coverage (61 tests)
A: Seed-only | B: Properties-only | C: Seed+Properties | D: Propagator V1 | E: Propagator V2 | F: Error cases | G: Edge cases

### CODE REVIEW

File: sdk/core/System.ClientModel/src/ModelReaderWriter/JsonPatch.Iterate.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        if (arrayBlob.IsEmpty)
+        {
+            ThrowKeyNotFoundException(jsonPath);
+        }
+
+        byte firstByte = arrayBlob.Span[0];
+        if (firstByte != (byte)'[')
+        {
+            throw new InvalidOperationException("The value at the specified path is not a JSON array.");
+        }
```
[the remark this answers is not in the document]
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/JsonPatch.Iterate.cs]: Good catch — the `firstByte` check was redundant since the `ArrayEnumerator` constructor already validates via `Utf8JsonReader` (which properly handles leading whitespace). Removed the byte check entirely and let the enumerator handle validation.

---
🤖 m-nash-copilot

File: sdk/core/System.ClientModel/src/ModelReaderWriter/JsonPatch.Iterate.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        // and seed resolution. Then post-process for sub-path removes/replacements.
+        ReadOnlyMemory<byte> resolvedArray;
+        if (TryGetEncodedValueInternal(jsonPath, out var ev) && ev.Kind != ValueKind.Removed)
+        {
+            resolvedArray = ev.Value;
+        }
+        else if (jsonPath.IsRoot())
+        {
+            resolvedArray = _rawJson.Value;
+        }
```
[the remark this answers is not in the document]
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/JsonPatch.Iterate.cs]: Good catch — restructured `ResolveArray` to explicitly check for `ValueKind.Removed` and throw `KeyNotFoundException` instead of falling through to the seed fallback. Also added test coverage for removed root/nested arrays and root non-array types (object, primitive).

---
🤖 m-nash-copilot

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/AvailabilitySetDataV2.Serialization.cs [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<AvailabilitySetDataV2>)this).GetFormatFromOptions(options) : options.Format;
+            switch (format)
+            {
+                case "J":
+                    using (JsonDocument document = JsonDocument.Parse(data))
+                        return DeserializeAvailabilitySetDataV2(document.RootElement, options, data);
+                default:
+                    throw new FormatException($"The model {nameof(AvailabilitySetDataV2)} does not support reading '{options.Format}' format.");
+            }
```
[the remark this answers is not in the document]
Comment (m-nash) [reply sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/AvailabilitySetDataV2.Serialization.cs:273]: Fixed both the read and write paths — both exception messages now use the computed `format` variable instead of `options.Format`, so they'll show the actual resolved format rather than 'W'.

---
🤖 m-nash-copilot

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/AvailabilitySetDataV2.Serialization.cs [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+        BinaryData IPersistableModel<AvailabilitySetDataV2>.Write(ModelReaderWriterOptions options)
+        {
+            var format = options.Format == "W" ? ((IPersistableModel<AvailabilitySetDataV2>)this).GetFormatFromOptions(options) : options.Format;
+            switch (format)
+            {
+                case "J":
+                    return ModelReaderWriter.Write(this, options, TestClientModelReaderWriterContext.Default);
+                default:
+                    throw new FormatException($"The model {nameof(AvailabilitySetDataV2)} does not support writing '{options.Format}' format.");
```
[the remark this answers is not in the document]
Comment (m-nash) [reply sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/AvailabilitySetDataV2.Serialization.cs]: Already fixed in the same commit as the read-side fix above — both paths now use the computed `format` variable.

---
🤖 m-nash-copilot

File: sdk/core/System.ClientModel/src/ModelReaderWriter/JsonPatch.Iterate.cs [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+            if (!_rawJson.Value.TryGetJson(jsonPath, out var result))
+            {
+                ThrowKeyNotFoundException(jsonPath);
+            }
+
+            return result;
+        }
+
+        // Case 3: Has properties and/or propagators — use TryGetEncodedValueInternal which
```
Comment (jsquire) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/JsonPatch.Iterate.cs:70]: These are quite helpful.  Thanks for including them!

File: sdk/core/System.ClientModel/tests/ModelReaderWriterTests/JsonPatchIterateTests.cs [hunk 6]
```diff
@@ -0,0 +1,1090 @@
...
+            var inner = Collect(jp, "$.m[0][0]"u8);
+            Assert.AreEqual(2, inner.Count);
+            Assert.AreEqual("1", inner[0]);
+            Assert.AreEqual("2", inner[1]);
+        }
+
+        [Test]
+        public void Seed_ObjectsWithNestedArrays()
+        {
+            JsonPatch jp = new("{\"arr\":[{\"tags\":[\"a\",\"b\"],\"id\":1},{\"tags\":[\"c\"],\"id\":2}]}"u8.ToArray());
```
Comment (jsquire) [inline sdk/core/System.ClientModel/tests/ModelReaderWriterTests/JsonPatchIterateTests.cs:137]: So \much \esca\ping fo\r \thi\s!

**Review by jsquire — APPROVED** [review]
We should add a change log entry here too, otherwise I fear that we'll forget later.

**Review by jsquire — APPROVED** [review]
